x-pack/legacy/plugins/code/server/lsp/abstract_launcher.ts

```diff
@@ -161,6 +154,33 @@ export abstract class AbstractLauncher implements ILanguageServerLauncher {
     }
   }
 
+  private async doSpawnProcess(
+    installationPath: string,
+    port: number,
+    log: Logger
+  ): Promise<ChildProcess> {
+    const child = await this.spawnProcess(installationPath, port, log);
+    const pid = child.pid;
+    this.currentPid = pid;
+    this.log.debug('spawned a child process ' + pid);
+    this.spawnTimes += 1;
+    this.startTime = Date.now();
+    this.running = true;
+    if (this.options.lsp.oomScoreAdj && process.platform === 'linux') {
+      try {
+        // clone form https://github.com/elastic/ml-cpp/blob/4dd90fa93338667b681364657222715f81c9868a/lib/core/CProcessPriority_Linux.cc
+        fs.writeFileSync(`/proc/${pid}/oom_score_adj`, `${OOM_SCORE_ADJ}\n`);
+        this.log.debug(`wrote oom_score_adj of process ${pid} to ${OOM_SCORE_ADJ}`);
+        fs.writeFileSync(`/proc/${pid}/oom_adj`, `${OOM_ADJ}\n`);
+        this.log.debug(`wrote oom_adj of process ${pid} to ${OOM_ADJ}`);
+      } catch (e) {
+        this.log.error('write oom_score_adj failed');
+        this.log.error(e);
+      }
```

Comment:
This is not doing exactly what the ML code is doing.

The idea of the ML code is:

1. First try `/proc/${pid}/oom_score_adj`
2. _Only_ if that fails, try writing to `/proc/${pid}/oom_adj`
3. _Only_ if that fails too log an error

The reasoning behind that is that newer kernels support `oom_score_adj`, older ones support `oom_adj`, and others support both.  On the kernels that support both if you write to one of these "files" it will change the value of the other one too.  For example:

```
[dave@linux elasticsearch]$ cat /proc/18965/oom_score_adj
0
[dave@linux elasticsearch]$ cat /proc/18965/oom_adj
0
[dave@linux elasticsearch]$ echo 500 > /proc/18965/oom_score_adj
[dave@linux elasticsearch]$ cat /proc/18965/oom_score_adj
500
[dave@linux elasticsearch]$ cat /proc/18965/oom_adj
7
[dave@linux elasticsearch]$ echo 12 > /proc/18965/oom_adj
[dave@linux elasticsearch]$ cat /proc/18965/oom_score_adj
705
[dave@linux elasticsearch]$ cat /proc/18965/oom_adj
12
```

I suspect what you currently have will:

1. Be OK on a kernel that supports both options
2. Log an unnecessary error on kernels that support `oom_score_adj` but not `oom_adj`
3. Fail to change the OOM adjustment on kernels that support `oom_adj` but not `oom_score_adj`

Originally `oom_adj` was due to be removed in August 2012, but it's hung around for a lot longer, so I suspect that most people are using kernels that fall under "Be OK on a kernel that supports both options".  However, trying to set it does spam the syslog with a deprecation warning, for example on Ubuntu 16.04 I get:

```
Aug  8 13:43:05 vm-ubuntu-1604-64-1 kernel: [8386972.848313] bash (29505): /proc/29505/oom_adj is deprecated, please use /proc/29505/oom_score_adj instead.
```

If you wanted to really do the same as ML the code would be more like this:

```
    try {
      // clone form https://github.com/elastic/ml-cpp/blob/4dd90fa93338667b681364657222715f81c9868a/lib/core/CProcessPriority_Linux.cc
      fs.writeFileSync(`/proc/${pid}/oom_score_adj`, `${OOM_SCORE_ADJ}\n`);
      this.log.debug(`wrote oom_score_adj of process ${pid} to ${OOM_SCORE_ADJ}`);
    } catch (e1) {
      try {
        fs.writeFileSync(`/proc/${pid}/oom_adj`, `${OOM_ADJ}\n`);
        this.log.debug(`wrote oom_adj of process ${pid} to ${OOM_ADJ}`);
	    } catch (e2) {
	      this.log.error('writing both oom_score_adj and oom_adj failed');
	      this.log.error(e1);
	      this.log.error(e2);
      }
    }
```

Reply:
Learned,  thank you.